token-upgrade/program/src/instruction.rs

```diff
@@ -0,0 +1,30 @@
+//! Program instructions
+
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+
+/// Instructions supported by the TokenUpgrade program
+#[derive(Clone, Copy, Debug, TryFromPrimitive, IntoPrimitive)]
+#[repr(u8)]
+pub enum TokenUpgradeInstruction {
+    /// Burns all of the source tokens in the user's account, and transfers the same
+    /// amount of tokens from an account owned by a PDA into another account.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writeable]` Source token account to burn from
+    ///   1. `[writeable]` Source token mint
+    ///   2. `[writeable]` Destination token upgrade account, owned by:
+    ///       `get_token_upgrade_authority_address(source_mint, destination_mint, program_id)`
```

Comment:
```suggestion
    ///   2. `[writeable]` Fount of destination tokens held in program escrow
```

Reply:
I found the term "token upgrade account" a little confusing so I'm trying to work around that a little here

Reply:
I had a hard time with naming... "Fount" might imply that it's generating tokens, how about just "bag"?